profiles/views.py

```diff
@@ -96,33 +95,47 @@ def get_queryset(self, pk=None):
         if self.request.method in ['PUT', 'PATCH']:
             return Profile.objects.filter(profile_id=pk)
 
+    def get_serializer_class(self):
+        get_contacts = self.request.query_params.get("get_contacts")
+        if self.request.method == 'GET':
+            return ProfileSensitiveDataROSerializer if get_contacts else ProfileDetailSerializer
+        else:
+            return ProfileSerializer
+
+    def retrieve(self, request, *args, **kwargs):
+        instance = self.get_object()
+        if not request.user.is_authenticated and request.query_params.get("get_contacts"):
+            return Response(status=status.HTTP_401_UNAUTHORIZED)
```

Comment:
Permissions must be checked in permission classes.

Reply:
That's a temporary solution until we have permission for the owner created.

Reply:
Permission is added and this is fixed now.